src/services/data-management/internal-mqtt-client-listener.service.ts

```diff
@@ -0,0 +1,83 @@
+﻿import {
+    forwardRef,
+    Inject,
+    Injectable,
+    Logger,
+    OnApplicationBootstrap,
+} from "@nestjs/common";
+import { ReceiveDataService } from "@services/data-management/receive-data.service";
+import { Client, connect } from "mqtt";
+import { IoTDeviceService } from "@services/device-management/iot-device.service";
+import { MQTTSubscriberDevice } from "@entities/mqtt-subscriber-device.entity";
+import { IoTDeviceType } from "@enum/device-type.enum";
+import { MqttClientId } from "@config/constants/mqtt-constants";
+import { EncryptionHelperService } from "@services/encryption-helper.service";
+
+@Injectable()
+export class InternalMqttClientListenerService implements OnApplicationBootstrap {
+    constructor(
+        private receiveDataService: ReceiveDataService,
+        @Inject(forwardRef(() => IoTDeviceService))
+        private iotDeviceService: IoTDeviceService,
+        private encryptionHelperService: EncryptionHelperService
+    ) {
+        this.clientDictionary = new Map<number, Client>();
+    }
+
+    clientDictionary: Map<number, Client>;
+    private readonly logger = new Logger(InternalMqttClientListenerService.name);
+
+    public async onApplicationBootstrap(): Promise<void> {
+        // Get all subscriber devices
+        const mqttSubscribers = await this.iotDeviceService.getAllMQTTSubscribers();
+
+        // Create clients for each device (Could share on a broker level)
+        this.createMQTTClients(
+            mqttSubscribers.filter(d => !this.clientDictionary.has(d.id))
+        );
+    }
+
+    public createMQTTClients(mqttSubscribers: MQTTSubscriberDevice[]) {
+        mqttSubscribers.forEach(d => {
+            const client = connect(d.mqttURL, {
```

Comment:
Can we make some logic that makes sure, that if the mqtt client can't connect because of a error, then it should not try again? Maybe set some flag in the db?